src/libraries/System.Private.CoreLib/src/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -218,6 +220,12 @@ public override void Return(T[] array, bool clearArray = false)
             if (log.IsEnabled())
             {
                 log.BufferReturned(array.GetHashCode(), array.Length, Id);
+                if (!(haveBucket & returned))
+                {
+                    log.BufferDropped(array.GetHashCode(), array.Length, Id,
```

Comment:
We are calculating `array.GetHashCode()` twice in a row here. Is that intentional, because the 2nd time is essentially free (gets it off the syncblock)?